hyperlink/_url.py

```diff
@@ -1464,3 +1504,329 @@ def remove(self, name):
+        _q = [tuple(_percent_decode(_encode_query_part(x),
+                                    raise_subencoding_exc=True)
+                    if x is not None else None
+                    for x in (k, v))
+              for k, v in self._url.query]
+        self._query = tuple(_q)
+        return self._query
+
+    @property
+    def fragment(self):
+        try:
+            return self._fragment
+        except AttributeError:
+            pass
+        frag = self._url.fragment
+        self._fragment = _percent_decode(_encode_fragment_part(frag),
+                                         raise_subencoding_exc=True)
+        return self._fragment
+
+    @property
+    def userinfo(self):
+        try:
+            return self._userinfo
+        except AttributeError:
+            pass
+        self._userinfo = tuple([_percent_decode(_encode_userinfo_part(p),
+                                                raise_subencoding_exc=True)
+                                for p in self._url.userinfo.split(':', 1)])
+        return self._userinfo
+
+    @property
+    def user(self):
+        return self.userinfo[0]
+
+    @property
+    def uses_netloc(self):
+        return self._url.uses_netloc
+
+    def replace(self, scheme=_UNSET, host=_UNSET, path=_UNSET, query=_UNSET,
+                fragment=_UNSET, port=_UNSET, rooted=_UNSET, userinfo=_UNSET,
+                uses_netloc=_UNSET):
+        """While the signature is the same, this `replace()` differs a little from
+        URL.replace. For instance, it accepts userinfo as a tuple, not
+        as a string. As with the rest of the methods on DecodedURL, if
+        you pass a reserved character, it will be automatically
+        encoded instead of an error being raised.
+        """
+        if path is not _UNSET:
+            path = [_encode_reserved(p) for p in path]
+        if query is not _UNSET:
+            query = [[_encode_reserved(x)
+                      if x is not None else None
+                      for x in (k, v)]
+                     for k, v in iter_pairs(query)]
+        if userinfo is not _UNSET:
+            if len(userinfo) > 2:
+                raise ValueError('userinfo expected sequence of ["user"] or'
+                                 ' ["user", "password"], got %r' % userinfo)
+            userinfo = u':'.join([_encode_reserved(p) for p in userinfo])
+        new_url = self._url.replace(scheme=scheme,
+                                    host=host,
+                                    path=path,
+                                    query=query,
+                                    fragment=fragment,
+                                    port=port,
+                                    rooted=rooted,
+                                    userinfo=userinfo,
+                                    uses_netloc=uses_netloc)
+        return type(self)(url=new_url)
+
+    def get(self, name):
+        "Get the value of all query parameters whose name matches *name*"
```

Comment:
Why not copy the docstring from `URL`?

Reply:
started with that, but most docstrings mention URL by name.